src/poetry/inspection/lazy_wheel.py

```diff
@@ -85,6 +89,24 @@ def metadata_from_wheel_url(
         # themselves are invalid, not because we've messed up our bookkeeping
         # and produced an invalid file.
         raise InvalidWheel(url, name)
+    except Exception as e:
+        if isinstance(e, LazyWheelUnsupportedError):
+            # this is expected when the code handles issues with lazy wheel metadata retrieval correctly
+            raise e
```

Comment:
I think it might be good to add a debug log for known exceptions because otherwise it might be difficult to tell why exactly lazy-wheel failed (e.g. no content-range).

Reply:
This will basically be raised as it previously used to. In this scenario we are simply re-raising the range/invalid wheel exception. Don't think the log would add anything meaningful. 

The catch-all is if it isn't a known error.